internal/controller/pipelinerollout/pipelinerollout_controller.go

```diff
@@ -1185,17 +1185,6 @@ func (r *PipelineRolloutReconciler) Recycle(ctx context.Context,
 
 	}
 
-	if requireDrain {
-		modified, err := r.ensurePipelineIsDrainable(ctx, pipeline)
-		if err != nil {
-			return false, err
-		}
-		if modified {
-			// since we just changed the pipeline, give numaflow a chance to reconcile it and then pause it, so requeue
-			return false, nil
-		}
-	}
```

Comment:
This code was making sure that if there were only 0 pods of a vertex, we bump it to 1 so it can drain.

Note: this is actually logic we should have been executing in the case of `if pause` and not just in the case of `if requireDrain` (we would want to be able to drain it if we're trying to pause at all, regardless of whether we needed to drain it)